royalties-registry/contracts/providers/RoyaltiesProviderArtBlocks.sol

```diff
@@ -23,37 +23,72 @@ contract RoyaltiesProviderArtBlocks is IRoyaltiesProvider, Ownable {
         //gettign artist and additionalPayee royalty part
         (address artistAddress, address additionalPayee, uint256 additionalPayeePercentage, uint256 royaltyFeeByID) = artBlocks.getRoyaltyData(tokenId);
 
+        require(additionalPayeePercentage <= 100 && royaltyFeeByID <= 100, "wrong additionalPayeePercentage value");
+
         //resulting royalties
         LibPart.Part[] memory result;
 
-        if (additionalPayeePercentage > 0){
+        //if no artist royalty
+        if (royaltyFeeByID == 0) {
+            //if artblocksPercentage > 0
+            if (artblocksPercentage > 0) {
+                result = new LibPart.Part[](1);
+
+                //calculating artBLocks part
+                result[0].account = payable(owner());
+                result[0].value = artblocksPercentage;
+            }
+            //if artblocksPercentage = 0 then result is empty
+            return result;
+
+        //if royaltyFeeByID > 0 and  0 < additionalPayeePercentage < 100
+        } else if (additionalPayeePercentage > 0 && additionalPayeePercentage < 100){
             result = new LibPart.Part[](3);
-        } else {
-            result = new LibPart.Part[](2);
-        }
 
-        //calculating artBLocks part
-        result[0].account = payable(owner());
-        result[0].value = artblocksPercentage;
+            //calculating artBLocks part
+            result[0].account = payable(owner());
+            result[0].value = artblocksPercentage;
 
-        // additional payee percentage * 100
-        uint96 additionalPart = uint96(royaltyFeeByID.mul(100).bp(additionalPayeePercentage.mul(100)));
+             // additional payee percentage * 100
+            uint96 additionalPart = uint96(royaltyFeeByID.mul(100).bp(additionalPayeePercentage.mul(100)));
 
-        //artist part
-        result[1].account = payable(artistAddress);
-        result[1].value = uint96(royaltyFeeByID.mul(100)) - additionalPart;
+            //artist part
+            result[1].account = payable(artistAddress);
+            result[1].value = uint96(royaltyFeeByID.mul(100)) - additionalPart;
 
-        //additional payee part
-        if (additionalPayeePercentage > 0) {
             result[2].account = payable(additionalPayee);
             result[2].value = additionalPart;
-        }
+            
+        //if royaltyFeeByID > 0 and additionalPayeePercentage == 0 or 100
+        } else {
+            result = new LibPart.Part[](2);
+
+            //calculating artBLocks part
+            result[0].account = payable(owner());
+            result[0].value = artblocksPercentage;
+
+            // additional payee percentage * 100
+            uint96 additionalPart = uint96(royaltyFeeByID.mul(100).bp(additionalPayeePercentage.mul(100)));
+
+            //artist part
+            if (additionalPayeePercentage == 0) {
+                result[1].account = payable(artistAddress);
+                result[1].value = uint96(royaltyFeeByID.mul(100)) - additionalPart;
```

Comment:
line 76 instead of "-" maybe more safy to us  sub?

Reply:
fixed